### PR TITLE
Small nits in service package

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: service/service.go [hunk 1]
```diff
@@ -43,10 +43,6 @@ type service struct {
 }
 
 func newService(set *svcSettings) (*service, error) {
-	if err := set.Config.Validate(); err != nil {
-		return nil, fmt.Errorf("invalid configuration: %w", err)
-	}
-
```
Comment (dmitryax) [inline service/service.go:49]: Is this a duplicate? Can you please point where it's being called?

**Review by dmitryax — COMMENTED** [review]
It would be nice to list updates in the commit/PR description?

File: service/service.go (hunk 1 above)
Comment (bogdandrutu) [reply service/service.go:49]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/service/config_watcher.go#L49

Comment (dmitryax) [reply service/service.go:49]: Thanks

**Review by dmitryax — APPROVED** [review]

**Pull request merged**